Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -257,6 +264,43 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
                                resourcesBuildPhase: resourcesBuildPhase)
     }
 
+    func generateCopyFilesBuildPhase(path: AbsolutePath,
+                                     target: Target,
+                                     graphTraverser: GraphTraversing,
+                                     pbxTarget: PBXTarget,
+                                     fileElements: ProjectFileElements,
+                                     pbxproj: PBXProj) throws
+    {
+        try target.copyFiles.forEach { phase in
+            let copyFilesPhase =  PBXCopyFilesBuildPhase(
+                dstPath: phase.subpath,
+                dstSubfolderSpec: PBXCopyFilesBuildPhase.SubFolder.init(rawValue: phase.destination.rawValue),
+                name: phase.name)
+
+            pbxproj.add(object: copyFilesPhase)
+            pbxTarget.buildPhases.append(copyFilesPhase)
+
+            var buildFilesCache = Set<AbsolutePath>()
+            let filePaths = phase.files
+                .map(\.path)
+                .sorted()
+                .cleanPackages()
```

Comment:
I've changed the `isPackage` check so it does not perform any disk check, instead it will check the UTI based of the file extension, that's reliable for testing. That is what `cleanPackages` uses to check and also that's only being executed once, just like `generateResourcesBuildFile` for removing files like .lproj and .xcassets.

Reply:
Awesome, thanks for the update!

The concern I had was if in inconsistency occurs between `ProjectFileElements` and `BuildPhaseGenerator`. `ProjectFileElements` currently constructs groups and elements based on the path. 

For example if it encounters the path:
`/Project/Templates/MyTemplate.template/manifest.json`

Given the project is in `/Project` the following will be created:

- Group `Templates`
- Group `Templates/MyTemplate.template`
- File Element `Templates/MyTemplate.template/manifest.json`

As such when `BuildPhaseGenerator` asks for a file element for `/Project/Templates/MyTemplate.template` it may fail seeing `/Project/Templates/MyTemplate.template` is a group. 

Interestingly though from testing this PR it doesn't seem we hit this issue - looks like the internal sorting of paths results in the package path `/Project/Templates/MyTemplate.template` being encountered before any of its nested files resulting in the file element for it getting created and subsequently paths for the nested files get dropped.

I guess it may become an issue later on if that behaviour changes - but at least now there is a fixture with acceptance tests that can help us catch regressions if anything changes 👍

Reply:
I got that! That makes sense, I've made the changes and now `cleanPackages` happens inside `CopyFilesAction+ManifestMapper` right after the globs are resolved and turned into paths.